app/models/manageiq/providers/amazon/cloud_manager/event_catcher/stream.rb

```diff
@@ -70,63 +57,87 @@ def each
-        queue = sqs.queues.create(queue_name)
-
+        queue_url = sqs_create_queue(queue_name)
         $aws_log.info("#{log_header} Subscribing Queue #{queue_name} to AWSConfig_topic")
-        aws_config_topic.subscribe(queue)
+        subscribe_topic_to_queue(sns_topic, queue_url)
         $aws_log.info("#{log_header} Created Amazon SQS Queue #{queue_name} and subscribed to AWSConfig_topic")
+        queue_url
       else
-        $aws_log.warn("#{log_header} Unable to find the AWS Config Topic. " \
-                      "Cannot collect Amazon events for AWS Access Key ID #{@aws_access_key_id}")
+        $aws_log.warn("#{log_header} Unable to find the AWS Config Topic '#{@topic_name}'. " \
+                      "Cannot collect Amazon events for AWS Access Key ID #{@ems.authentication_userid}")
         $aws_log.warn("#{log_header} Contact Amazon to create the AWS Config service and topic for Amazon events.")
-        queue = nil
-        raise ProviderUnreachable
+        raise ProviderUnreachable, "Unable to find the AWS Config Topic '#{@topic_name}'"
       end
+    rescue Aws::SQS::Errors::ServiceError => exception
+      raise ProviderUnreachable, exception.message
     end
-    queue
   end
 
-  def find_aws_config_topic
-    sns.topics.detect { |t| t.name == @topic_name }
+  def sqs_create_queue(queue_name)
+    @ems.with_provider_connection(:service => :SQS, :sdk_v2 => true) do |sqs|
+      sqs.client.create_queue(:queue_name => queue_name).queue_url
+    end
   end
 
-  def collect_events(queue)
-    events = []
-    # http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/SQS/Queue.html
-    # :batch_size   - maximum number of messages to retrieve per request
-    # :idle_timeout - maximum number of seconds to poll while no messages are returned
-    queue.poll(:idle_timeout => 5) do |amazon_message|
-      sns_message = amazon_message.as_sns_message
+  def sqs_get_queue_url(queue_name)
+    @ems.with_provider_connection(:service => :SQS, :sdk_v2 => true) do |sqs|
+      sqs.client.get_queue_url(:queue_name => queue_name).queue_url
+    end
+  end
 
-      event = parse_event(sns_message)
-      events << event if event
+  # @return [Aws::SNS::Topic] the found topic or nil
+  def find_sns_topic(topic_name)
+    @ems.with_provider_connection(:service => :SNS, :sdk_v2 => true) do |sns|
+      sns.topics.detect { |t| t.arn.split(/:/)[-1] == topic_name }
     end
-    events
   end
 
-  def parse_event(sns_message)
+  # @param [Aws::SNS::Topic] aws_config_topic
+  # @param [String] queue_url
+  def subscribe_topic_to_queue(aws_config_topic, queue_url)
+    # the old behavior also added a policy to the queue to allow the topic
+    # to send the queue messages - not sure if this is wanted...
+    # https://github.com/aws/aws-sdk-ruby/blob/74ba5e/lib/aws/sns/topic.rb#L86-L90
+    # https://github.com/aws/aws-sdk-ruby/blob/74ba5e/lib/aws/sns/topic.rb#L368-L378
```

Comment:
Right.  The topic definitely needs to be able to publish to the queue so the messages can flow back to the listener on our side.

What does that mean for this code?  Do you have to change the implementation?  Or, does that become a documentation issue?  It would be nice if the code could do it.